### ISSUE TITLE
Why _refresh and invalidate doesn't work?

### ISSUE THREAD

**Issue by wsf1990** [peterbe/django-cache-memoize#30]
I'm using this package in django,like this:

```
class A(models.Model):
    def b(self, arg1='', arg2=None)
         return ...
```
And clear cache like this:
```
self.b(_refresh=True)
self.b.invalidate(self)
```
Both this two method can't clear the method b's cache.

**Comment by wsf1990** [comment]
I found that

**Comment by akx** [comment]
Your example doesn't have any function decorated with `@cache_memoize()`...

**Comment by wsf1990** [comment]
I'm sorry,it's my mistake.

**Comment by wsf1990** [comment]
Now the question is that,When I want to clear the cache,I must call the method with `_refresh=True` without positional arg name,like this:
```
self.b('', None, _refresh=True)
```
if call this method with positional arg name, the clear will not work.Like this:
```
self.b(arg1='', arg2=None, _refresh=True)
```

**Comment by wsf1990** [comment]
And how can I clear a function's cache by all args?Like function b's arg1 equals 1,2,3,4,and so on, and arg2 equals '1', '2', and so on.

**Comment by peterbe** [comment]
> if call this method with positional arg name, the clear will not work

Correct. The invalidation signature needs to the exact same way as the non-invalidation signature.

**Comment by peterbe** [comment]
> But I also want to what can I do to practice this.

Practice? Can you try to phrase yourself differently?

**Comment by wsf1990** [comment]
I'm sorry.What I mean is that how can I clear all cache of a function.Do you have any advise?

**Comment by akx** [comment]
With the current implementation, there's no way to clear all cache for a function, unless you specify a custom `prefix` and change it every time you need to clear the cache.
